plugin/trino-memory/src/main/java/io/trino/plugin/memory/MemoryMetadata.java

```diff
@@ -174,7 +175,7 @@ public synchronized ConnectorTableMetadata getTableMetadata(ConnectorSession ses
         MemoryTableHandle handle = (MemoryTableHandle) tableHandle;
         return tables.get(handle.getId())
                 .getColumns().stream()
-                .collect(toMap(ColumnInfo::getName, ColumnInfo::getHandle));
```

Comment:
As I understand `toMap` is used here for something else, can you please remove all occurrences of `toMap`.

Reply:
But since we pass this to `Trino` it can be immutable I guess